layout/_partials/head/head-unique.swig

```diff
@@ -26,3 +26,10 @@
 {% if theme.canonical %}
   <link rel="canonical" href="{{ config.url }}/{{ page.canonical_path.replace('index.html', '') }}"/>
 {% endif %}
+
+{# Exports some front-matter variables to Front-End #}
+<script type="text/javascript" id="page.configurations">
+  CONFIG.page = {
+    sidebar: {{ page.sidebar | json_encode }},
```

Comment:
how about `sidebar: {{ page.title | page.sidebar | json_encode }},` ?
make sure the sidebar diaplay in post with titile when `CONFIG.sidebar.display === 'post' || CONFIG.sidebar.display === 'always'`, don't need to add `sidebar: true` in front matter of posted pages.

Reply:
That will make a different change with the original logic. The original logic is the `sidebar.display === 'post' || sidebar.display === 'always'` **AND** the page has a TOC in the meanwhile.

Now the code is only add an additional option that allow the user to change it. I think it is better without a breaking change. 😃